### PR TITLE
dpdk: count number of packets with invalid checksum v3

### PR DESCRIPTION
**Pull request by lukashino**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/5553) ticket:

Follow up of #7900  

Describe changes:
- xstats output

### CODE REVIEW

File: src/source-dpdk.c [hunk 1]
```diff
@@ -491,6 +509,50 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
     SCReturnInt(TM_ECODE_FAILED);
 }
 
+static void PrintDPDKPortXstats(uint32_t port_id, const char *port_name)
+{
+    int len, ret, i;
+    struct rte_eth_xstat *xstats;
+    struct rte_eth_xstat_name *xstats_names;
+    static const char *stats_border = "_______";
```
Comment (victorjulien) [inline src/source-dpdk.c:517]: stale

File: src/source-dpdk.c [hunk 2]
```diff
@@ -364,20 +364,38 @@ static TmEcode ReceiveDPDKLoop(ThreadVars *tv, void *data, void *slot)
...
+                if ((ol_flags & RTE_MBUF_F_RX_IP_CKSUM_MASK) == RTE_MBUF_F_RX_IP_CKSUM_GOOD &&
+                        (ol_flags & RTE_MBUF_F_RX_L4_CKSUM_MASK) == RTE_MBUF_F_RX_L4_CKSUM_GOOD) {
+                    SCLogDebug("HW detected GOOD IP and L4 chsum, skipping...");
+                    p->flags |= PKT_IGNORE_CHECKSUM;
+                } else if ((ol_flags & RTE_MBUF_F_RX_IP_CKSUM_MASK) == RTE_MBUF_F_RX_IP_CKSUM_BAD ||
+                           (ol_flags & RTE_MBUF_F_RX_L4_CKSUM_MASK) == RTE_MBUF_F_RX_L4_CKSUM_BAD) {
+                    SCLogDebug("HW detected BAD IP or L4 chsum, discarding pkt...");
+                    (void)SC_ATOMIC_ADD(ptv->livedev->invalid_checksums, 1);
+
+                    // bad checksum, releasing the packet
```
Comment (victorjulien) [inline src/source-dpdk.c:393]: we should still process the packet, so we can match on it with the csum keyword

File: src/source-dpdk.c [hunk 3]
```diff
@@ -491,6 +509,50 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
     SCReturnInt(TM_ECODE_FAILED);
 }
 
+static void PrintDPDKPortXstats(uint32_t port_id, const char *port_name)
+{
+    int len, ret, i;
```
Comment (victorjulien) [inline src/source-dpdk.c:514]: declare vars on first use where possible

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/source-dpdk.c (hunk 1 above)
Comment (lukashino) [reply src/source-dpdk.c:517]: uh oh sorry

File: src/source-dpdk.c (hunk 2 above)
Comment (lukashino) [reply src/source-dpdk.c:393]: done

File: src/source-dpdk.c (hunk 3 above)
Comment (lukashino) [reply src/source-dpdk.c:514]: done

**Comment by lukashino** [comment]
Follow up #7947

**Pull request closed without merging**